### DISCUSSION TITLE
Cannot install necessary programs to setup Raspberry Pi. Always getting errors.

### DISCUSSION THREAD

**Discussion by ritog** [microsoft/IoT-For-Beginners#377] [Q&A]
I am trying to use a Raspberry Pi as a headless device, and while I am connected to it through ssh, I cannot complete the suggested setup.

I am trying to follow instruction from [the Raspberry Pi page](https://github.com/ghosh-r/IoT-For-Beginners/blob/main/1-getting-started/lessons/1-introduction-to-iot/pi.md#remote-access-to-code-the-pi).

There, the instruction is to run a full upgrade and then run these commands:

```sh
curl -sL https://github.com/Seeed-Studio/grove.py/raw/master/install.sh | sudo bash -s -
```
But, when I run this command, I get thrown a lot of errors, the last line of which is:

```sh
-------------------------------------------------------
     Grove.py installation FAILED, FAILED, FAILED      
-------------------------------------------------------
```

And here is the full output:

<details>
  <summary>Click for full output!</summary>

```
 Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
Hit:1 https://seeed-studio.github.io/pi_repo stretch InRelease
Hit:2 http://raspbian.raspberrypi.org/raspbian bullseye InRelease
Hit:3 http://archive.raspberrypi.org/debian bullseye InRelease  
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-pip is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python3-pip

E: Package 'python-pip' has no installation candidate
dpkg-query: package 'python-pip' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-pip is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python3-pip

E: Package 'python-pip' has no installation candidate
dpkg-query: package 'python-pip' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-pip is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python3-pip

E: Package 'python-pip' has no installation candidate
dpkg-query: package 'python-pip' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3-pip is already the newest version (20.3.4-4+rpt1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
I2C interface enabled...
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-rpi.gpio is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python-rpi.gpio' has no installation candidate
dpkg-query: package 'python-rpi.gpio' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-rpi.gpio is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python-rpi.gpio' has no installation candidate
dpkg-query: package 'python-rpi.gpio' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package python-rpi.gpio is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python-rpi.gpio' has no installation candidate
dpkg-query: package 'python-rpi.gpio' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
-------------------------------------------------------
     Grove.py installation FAILED, FAILED, FAILED      
-------------------------------------------------------
```

</details>

Can you help me find a solution to this?

**Comment by ritog** [comment]
I have finished the rest of the setup:

![image](https://user-images.githubusercontent.com/58001482/169616567-685db69e-3688-4d5d-a94d-8bf9b2d0f867.png)

**Comment by Rohan03march** [comment]
On Sat, 21 May 2022 at 3:15 AM, Ritobrata Ghosh ***@***.***>
wrote:

> I have finished the rest of the setup:
>
> [image: image]
> <https://user-images.githubusercontent.com/58001482/169616567-685db69e-3688-4d5d-a94d-8bf9b2d0f867.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/IoT-For-Beginners/discussions/377#discussioncomment-2793949>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU2E2KQQL5ES4CLQLLQPN73VLABWRANCNFSM5WQUEQZQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <microsoft/IoT-For-Beginners/repo-discussions/377/comments/2793949@
> github.com>
>
Then what is the Problem that your facing With the setup

**Comment by Rohan03march** [comment]
On Tue, 31 May 2022 at 4:16 PM, Rohan Ronnie ***@***.***>
wrote:

>
>
> On Sat, 21 May 2022 at 3:15 AM, Ritobrata Ghosh ***@***.***>
> wrote:
>
>> I have finished the rest of the setup:
>>
>> [image: image]
>> <https://user-images.githubusercontent.com/58001482/169616567-685db69e-3688-4d5d-a94d-8bf9b2d0f867.png>
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/microsoft/IoT-For-Beginners/discussions/377#discussioncomment-2793949>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AU2E2KQQL5ES4CLQLLQPN73VLABWRANCNFSM5WQUEQZQ>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: <microsoft/IoT-For-Beginners/repo-discussions/377/comments/2793949@
>> github.com>
>>
> Then what is the Problem that your facing With the setup

mad I know it is better you kindly use resbarry pie

**Comment by jimbobbennett** [comment]
The Speed script only works on older Raspberry Pi OS versions. It depends on Python 2, but this is not installed on the latest Bullseye versions of Raspberry Pi OS.

You need to do this to install the Grove libraries:

```sh
sudo apt install git python3-dev python3-pip --yes

git clone https://github.com/Seeed-Studio/grove.py
cd grove.py
sudo pip3 install .

sudo raspi-config nonint do_i2c 0
```

I'm working through this at the moment updating all the out of date stuff, so this will be updated in the lesson in the next couple of days.
